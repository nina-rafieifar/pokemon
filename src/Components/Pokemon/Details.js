import React from "react";
import "../Attributes/Abilities";
import "../Attributes/Attacks";
import "./style.scss";

export function Details(prop) {
  return (
    <div className="pokemon__details_table">
      <pokemon-abilities resistant={prop.pokemon.resistant} weaknesses={prop.pokemon.weaknesses}></pokemon-abilities>
      <div className="pokemon__details_attacks">
        {prop.pokemon.attacks.fast && (
          <div className="chips">
            <div className="chip__text">Fast</div>
            {prop.pokemon.attacks.fast.map((i) => (
              <pokemon-attacks name={i.name} damage={i.damage} key={i.name}></pokemon-attacks>
            ))}
          </div>
        )}
        {prop.pokemon.attacks.special && (
          <div className="chips">
            <div className="chip__text">Special</div>
            <div>
              {prop.pokemon.attacks.special.map((i) => (
                <pokemon-attacks name={i.name} damage={i.damage} key={i.name}></pokemon-attacks>
              ))}
            </div>
          </div>
        )}
      </div>
    </div>
  );
}
