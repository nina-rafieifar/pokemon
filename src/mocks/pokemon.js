export const pokemon = {
  attacks: { fast: [{ damage: 12, name: "Tackle" }], special: [{ damage: 70, name: "Power Whip" }] },
  evolutionRequirements: { amount: 25, name: "Bulbasaur candies" },
  evolutions: [
    {
      id: "UG9rZW1vbjowMDI=",
      number: "002",
      name: "Ivysaur",
      image: "https://img.pokemondb.net/artwork/ivysaur.jpg"
    },
    {
      id: "UG9rZW1vbjowMDM=",
      number: "003",
      name: "Venusaur",
      image: "https://img.pokemondb.net/artwork/venusaur.jpg"
    }
  ],
  height: { __typename: "PokemonDimension", minimum: "0.61m", maximum: "0.79m" },
  id: "UG9rZW1vbjowMDE=",
  image: "https://img.pokemondb.net/artwork/bulbasaur.jpg",
  name: "Bulbasaur",
  number: "001",
  resistant: ["Water", "Electric", "Grass", "Fighting", "Fairy"],
  weaknesses: ["Fire", "Ice", "Flying", "Psychic"],
  weight: { minimum: "6.04kg", maximum: "7.76kg" }
};
