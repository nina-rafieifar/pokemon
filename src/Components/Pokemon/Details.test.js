import React from "react";
import { render } from "@testing-library/react";
import { Details } from "./Details";
import { pokemon } from "../../mocks/pokemon";

test("renders Pokemon", () => {
  const element = render(<Details pokemon={pokemon} />);
  expect(element).toMatchSnapshot();
});

test("renders Pokemon no attacks", () => {
  const noAttackPokemon = pokemon;
  noAttackPokemon.attacks.fast = null;
  noAttackPokemon.attacks.special = [];
  const element = render(<Details pokemon={noAttackPokemon} />);
  expect(element).toMatchSnapshot();
});
