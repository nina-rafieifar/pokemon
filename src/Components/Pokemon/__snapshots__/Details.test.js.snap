// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Pokemon 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pokemon__details_table"
      >
        <pokemon-abilities
          resistant="Water,Electric,Grass,Fighting,Fairy"
          weaknesses="Fire,Ice,Flying,Psychic"
        />
        <div
          class="pokemon__details_attacks"
        >
          <div
            class="chips"
          >
            <div
              class="chip__text"
            >
              Fast
            </div>
            <pokemon-attacks
              damage="12"
              name="Tackle"
            />
          </div>
          <div
            class="chips"
          >
            <div
              class="chip__text"
            >
              Special
            </div>
            <div>
              <pokemon-attacks
                damage="70"
                name="Power Whip"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pokemon__details_table"
    >
      <pokemon-abilities
        resistant="Water,Electric,Grass,Fighting,Fairy"
        weaknesses="Fire,Ice,Flying,Psychic"
      />
      <div
        class="pokemon__details_attacks"
      >
        <div
          class="chips"
        >
          <div
            class="chip__text"
          >
            Fast
          </div>
          <pokemon-attacks
            damage="12"
            name="Tackle"
          />
        </div>
        <div
          class="chips"
        >
          <div
            class="chip__text"
          >
            Special
          </div>
          <div>
            <pokemon-attacks
              damage="70"
              name="Power Whip"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders Pokemon no attacks 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pokemon__details_table"
      >
        <pokemon-abilities
          resistant="Water,Electric,Grass,Fighting,Fairy"
          weaknesses="Fire,Ice,Flying,Psychic"
        />
        <div
          class="pokemon__details_attacks"
        >
          <div
            class="chips"
          >
            <div
              class="chip__text"
            >
              Special
            </div>
            <div />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pokemon__details_table"
    >
      <pokemon-abilities
        resistant="Water,Electric,Grass,Fighting,Fairy"
        weaknesses="Fire,Ice,Flying,Psychic"
      />
      <div
        class="pokemon__details_attacks"
      >
        <div
          class="chips"
        >
          <div
            class="chip__text"
          >
            Special
          </div>
          <div />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
